packages/app-builder-lib/src/targets/snap/snapBuilder.ts

```diff
@@ -0,0 +1,187 @@
+      return await getBinFromUrl(
+        "snap-template",
+        "4.0-2",
+        "PYhiQQ5KE4ezraLE7TOT2aFPGkBNjHLRN7C8qAPaC6VckHU3H+0m+JA/Wmx683fKUT2ZBwo9Mp82EuhmQo5WOQ==",
+        "snap-template-electron-4.0-2-amd64.tar"
+      )
+    case "electron4:armhf":
+    case "electron4:arm":
+      return await getBinFromUrl(
+        "snap-template",
+        "4.0-1",
+        "jK+E0d0kyzBEsFmTEUIsumtikH4XZp8NVs6DBtIBJqXAmVCuNHcmvDa0wcaigk8foU4uGZXsLlJtNj11X100Bg==",
+        "snap-template-electron-4.0-1-armhf.tar"
+      )
+    default:
+      return await getBin("snap-template-custom", templateUrl, templateSha512)
+  }
+}
+
+async function snap(options: SnapBuilderOptions, templateDir?: string): Promise<void> {
+  const isUseTemplateApp = !isEmptyOrSpaces(templateDir)
+  const stageDir = options.stageDir
+  const snapMetaDir = isUseTemplateApp ? path.join(stageDir, "meta") : path.join(stageDir, "snap")
+
+  if (options.icon) {
+    await copyFile(options.icon, path.join(snapMetaDir, "gui", "icon" + path.extname(options.icon)))
+  }
+  if (options.hooksDir) {
+    await copyDir(options.hooksDir, path.join(snapMetaDir, "hooks"))
+  }
+
+  const scriptDir = path.join(stageDir, "scripts")
+  await mkdir(scriptDir, { recursive: true })
+
+  if (options.executableName) {
+    await writeCommandWrapper(options, isUseTemplateApp, scriptDir)
+  }
+
+  const chromeSandbox = path.join(options.appDir, "app", "chrome-sandbox")
+  if (await exists(chromeSandbox)) {
+    await unlink(chromeSandbox)
+  }
+
+  if (isUseTemplateApp) {
+    await buildUsingTemplate(templateDir, options)
+  } else {
+    await buildWithoutTemplate(options, scriptDir)
+  }
+}
+
+async function writeCommandWrapper(options: CommandWrapperOptions, isUseTemplateApp: boolean, scriptDir: string): Promise<void> {
+  const appPrefix = isUseTemplateApp ? "" : "app/"
+  const dir = isUseTemplateApp ? options.stageDir || "" : scriptDir
+
+  const commandWrapperFile = path.join(dir, "command.sh")
+  let text = `#!/bin/bash -e\nexec "$SNAP/desktop-init.sh" "$SNAP/desktop-common.sh" "$SNAP/desktop-gnome-specific.sh" "$SNAP/${appPrefix}${options.executableName}"`
+
+  if (options.extraAppArgs?.length) {
+    text += ` ${options.extraAppArgs.join(" ")}`
+  }
+  text += ' "$@"'
+
+  await writeFile(commandWrapperFile, text, { mode: 0o755 })
+  await chmod(commandWrapperFile, 0o755)
+}
+
+async function buildUsingTemplate(templateDir: string, options: SnapBuilderOptions): Promise<void> {
+  const stageDir = options.stageDir
+
+  // https://github.com/electron-userland/electron-builder/issues/3608
+  // even if electron-builder will correctly unset setgid/setuid, still, quite a lot of possibilities for user to create such incorrect permissions,
+  // so, just unset it using chmod right before packaging
+  const dirs = [stageDir, options.appDir, templateDir]
+  await Promise.all(
+    dirs
+      .filter(dir => !!dir)
+      .map(dir =>
+        exec(`chmod -R g-s ${dir}`).catch((err: any) => {
```

Comment:
How about using `fs.chmodSync`?

Reply:
Sure thing, I'l double check that. I was during purely a 1:1 migration from the app-builder-bin code, which was calling `chmod` directly, but I'll see if it's possible to leverage built-in fs functions here